docs: creating artefacts (#524)
DiamondLightSource/workflows: docs/how-tos/create-artifacts.md

> # Creating artifacts

Reviewer: ```suggestion
# Creating Artifacts
```

## Preface

> This guide will describe how to create artifacts and export them at the end of a workflow run. These artifacts are then made available on the dashboard where they can be downloaded.

Reviewer: ```suggestion
This guide will describe how to create artifacts and export them at the end of a workflow run. These artifacts are then made available on the [dashboard ](https://workflows.diamond.ac.uk/workflows) where they can be downloaded.
```

This workflow also has a local-volume mouted to it to store some files in.

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: mount-tmpdir
spec:
  entrypoint: workflow-entry
  volumeClaimTemplates:
  - metadata:
      name: tmpdir
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 1Gi
      storageClassName: local-path

  templates:
  - name: bash
    inputs:
      parameters:
      - name: command
    container:
      image: busybox
      volumeMounts:
      - name: tmpdir
        mountPath: /tmp
      command: ["/bin/sh", "-c"]
      args:
      - "{{ inputs.parameters.command }}"

  - name: workflow-entry
    dag:
      tasks:
      - name: say-hello
        template: bash
        arguments:
          parameters:
          - name: command
            value: |
              echo "This is my message!" > /tmp/some-text.txt
              cat /tmp/some-text.txt
```

## Modifying the Template

> We must modify the workflow template to expect to create an artifact. Add the `output` tag with the name and path of the expected artifact.

Reviewer: ```suggestion
We must modify the workflow template for it to create an artifact. Add the `output` tag with the name and path of the expected artifact.
```

```yaml
  - name: bash
    inputs:
      parameters:
      - name: command
    container:
      image: busybox
      volumeMounts:
      - name: tmpdir
        mountPath: /tmp
      command: ["/bin/sh", "-c"]
      args: - "{{ inputs.parameters.command }}"
    outputs:
      artifacts:
      - name: text
        path: /tmp/some-text.txt
        archive:
          none: { }
```

By default, all artifact are tarballs. This can be disabled by adding `none` to the archive, meaning artifact are saved as their raw type. Supported artifact types will be available for preview on the dashboard, but taraballs will have to be downloaded to view their content so we suggest not archiving small artifacts - such as images and text files.

> The complete workflow is provided below for reference.

Reviewer: Could use "??? example" here to collapse the full manifest ?
